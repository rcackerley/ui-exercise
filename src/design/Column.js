// @flow strict @format

import React from "react"
import type { Node } from "react"

import injectSheet from "react-jss"

import classNames from "classnames"

type Props = {|
  classes: { [string]: string },
  children: Node,
  centered: boolean,
  customStyles: string
|}

const styles = {
  column: {
    display: "flex",
    flexDirection: "column"
  },
  centered: {
    alignItems: "center"
  }
}

const Column = ({ classes, children, centered, customStyles }: Props) => (
  <div
    className={classNames(
      classes.column,
      centered && classes.centered,
      customStyles
    )}
  >
    {children}
  </div>
)

export default injectSheet(styles)(Column)
