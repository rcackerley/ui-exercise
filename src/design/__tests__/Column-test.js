import React from "react"
import ReactDOM from "react-dom"
import Column from "../Column"

import renderer from "react-test-renderer"

it("render Column component correctly", () => {
  const ColumnComponent = renderer.create(<Column />).toJSON()
  expect(ColumnComponent).toMatchSnapshot()
})
