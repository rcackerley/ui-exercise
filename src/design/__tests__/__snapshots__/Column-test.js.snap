// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Column component correctly 1`] = `
<div
  className="Column-column-0-1-1"
/>
`;
